include/roboteam_ai/world_new/WorldData.hpp

```diff
@@ -0,0 +1,139 @@
+//
+// Created by john on 12/16/19.
+//
+
+#ifndef RTT_WORLD_DATA_HPP
+#define RTT_WORLD_DATA_HPP
+
+#include <vector>
+
+#include "roboteam_proto/World.pb.h"
+#include "roboteam_proto/Setting.pb.h"
+#include "roboteam_proto/RobotFeedback.pb.h"
+
+#include "Settings/Settings.h"
+#include "world_new/views/RobotView.hpp"
+#include "world_new/views/BallView.hpp"
+
+namespace rtt::world_new {
+
+    namespace robot {
+        class Robot;
+    } // namespace robot
+
+    /**
+     * WorldData structs
+     * Hold data about a specific time point in the world.
+     * Holds:
+     *  Robots
+     *  Ball
+     *  Timepoint
+     */
+    class WorldData {
+        friend class World;
```

Comment:
Why use this friend class when you have your getters and setters?

Reply:
Default constructor gets exposed to World this way, I don't want the default ctor to be public.